src/lcd/nextion.cpp

```diff
@@ -29,11 +29,24 @@ void lcdWakeUp(void) {
   myNex.writeNum("sleep", 0);
 }
 
+// Don't need to upload this again after boot
+void lcdInitProfiles(eepromValues_t &eepromCurrentValues) {
+  // Highlight the active profile
+  myNex.writeNum("pIdx", eepromCurrentValues.activeProfile + 1  /* 1-offset in nextion */);
+  String component = "qPf" + (eepromCurrentValues.activeProfile + 1);
+  myNex.writeNum(component + "bco", myNex.readNumber(component + "bco2"));
+  myNex.writeNum(component + "pco", myNex.readNumber(component + "pco2"));
```

Comment:
These are handled in Nextion i can confirm.

Reply:
ah, I see it now - have you tested the execution order? isn't it possible the nextion mcu will go through the page init code before the system sets `pIdx`, or is there a synchronization point to guarantee that can’t happen? 

I see `lcdInit()` talks to the screen early on and sets the version, but the page advance seems to be handled independently by a timer.

I’ll remove it for now on the assumption the init code takes way shorter than the timer and we’re good. At some point we might want to make the page advance handled by the MCU (as a bonus, it could also be an indicator of no comms for install troubleshooting)

Reply:
The no comms part is already handled by the "ack" global var, there are certain parts which are a tell whether the init part is all good or not.

Reply:
We can chat in private more since it's a lengthier discussion.